File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Administration/Language/LanguageDialog.cs

namespace BasicApplication.Administration
{
    using jQueryApi;
    using Serenity;
    using System.Collections.Generic;

    [IdProperty("Id"), NameProperty("LanguageName")]
    [FormKey("Administration.Language"), LocalTextPrefix("Administration.Language"), Service("Administration/Language")]
    public class LanguageDialog : EntityDialog<LanguageRow>, IAsyncInit
    {
    }
}

File: serene/src/Serene.Web/Modules/Administration/Language/LanguageDialog.ts
import { EntityDialog } from "@serenity-is/corelib";
import { LanguageForm, LanguageRow, LanguageService } from "../../ServerTypes/Administration";
import { nsAdministration } from "../../ServerTypes/Namespaces";

export class LanguageDialog extends EntityDialog<LanguageRow, any> {
    static override[Symbol.typeInfo] = this.registerClass(nsAdministration);

    protected override getFormKey() { return LanguageForm.formKey; }
    protected override getIdProperty() { return LanguageRow.idProperty; }
    protected override getLocalTextPrefix() { return LanguageRow.localTextPrefix; }
    protected override getNameProperty() { return LanguageRow.nameProperty; }
    protected override getService() { return LanguageService.baseUrl; }

    protected form = new LanguageForm(this.idPrefix);
}